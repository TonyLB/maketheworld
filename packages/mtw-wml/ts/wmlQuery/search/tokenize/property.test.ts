import SourceStream from '../../../parser/tokenizer/sourceStream'
import propertyTokenizer from './property'

describe('propertyTokenizer', () => {
    beforeEach(() => {
        jest.clearAllMocks()
        jest.resetAllMocks()
    })

    it('should return undefined when no match', () => {
        const testStream = new SourceStream('Testing')
        expect(propertyTokenizer(testStream)).toBe(undefined)
    })
    it('should tokenize property', () => {
        const testStream = new SourceStream('[key="test"] Test')
        expect(propertyTokenizer(testStream)).toMatchSnapshot()
    })
    it('should tokenize nthChild to end of source', () => {
        const testStream = new SourceStream('[key = "test" ]')
        expect(propertyTokenizer(testStream)).toMatchSnapshot()
        expect(testStream.isEndOfSource).toBe(true)
    })
})
