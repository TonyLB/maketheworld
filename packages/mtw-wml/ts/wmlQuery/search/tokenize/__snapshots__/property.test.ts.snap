// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propertyTokenizer should tokenize nthChild to end of source 1`] = `
Object {
  "endIdx": 14,
  "key": "key",
  "startIdx": 0,
  "type": "Property",
  "value": "test",
}
`;

exports[`propertyTokenizer should tokenize property 1`] = `
Object {
  "endIdx": 11,
  "key": "key",
  "startIdx": 0,
  "type": "Property",
  "value": "test",
}
`;
